test/unit/org/apache/cassandra/net/HandshakeTest.java

```diff
@@ -193,48 +193,31 @@ public void testSendCompatibleOldVersionPre40() throws InterruptedException
     }
 
     @Test
-    public void testSendIncompatibleOldVersionPre40() throws InterruptedException
+    public void testSendIncompatibleOldVersion40() throws InterruptedException
     {
         try
         {
-            handshake(VERSION_30, VERSION_30, VERSION_30, VERSION_3014, VERSION_3014);
-            Assert.fail("Should have thrown");
+            Assert.fail(Objects.toString(handshake(VERSION_30, VERSION_30, VERSION_30, current_version, current_version)));
         }
         catch (ExecutionException e)
         {
             assertTrue(e.getCause() instanceof ClosedChannelException);
         }
     }
 
-    @Test
-    public void testSendCompatibleOldVersion40() throws InterruptedException, ExecutionException
-    {
-        Result result = handshake(VERSION_30, VERSION_30, VERSION_30, VERSION_30, current_version);
-        Assert.assertEquals(Result.Outcome.SUCCESS, result.outcome);
-        Assert.assertEquals(VERSION_30, result.success().messagingVersion);
-    }
-
-    @Test
-    public void testSendIncompatibleOldVersion40() throws InterruptedException
+    @Test // fairly contrived case, but since we introduced logic for testing we need to be careful it doesn't make us worse
+    public void testSendToFuturePost40BelievedToBePre40() throws InterruptedException, ExecutionException
     {
         try
         {
-            Assert.fail(Objects.toString(handshake(VERSION_30, VERSION_30, VERSION_30, current_version, current_version)));
+            Assert.fail(Objects.toString(handshake(VERSION_30, VERSION_30, current_version, VERSION_30, current_version + 1)));
```

Comment:
Totally optional, but it can be simplified to:
```java
public void testSendToFuturePost40BelievedToBePre40()
{
    Assertions.assertThatThrownBy(() -> Objects.toString(handshake(VERSION_30, VERSION_30, current_version, VERSION_30, current_version + 1)))
              .hasRootCauseInstanceOf(ClosedChannelException.class);
}
```

Reply:
Same for `testSendCompatibleOldVersion40` and `testSendIncompatibleOldVersion40` right above.